lib/geoengineer/resources/aws/api_gateway/aws_api_gateway_method.rb

```diff
@@ -9,18 +9,32 @@ class GeoEngineer::Resources::AwsApiGatewayMethod < GeoEngineer::Resource
   include GeoEngineer::ApiGatewayHelpers
 
   validate -> {
-    validate_required_attributes([
-                                   :rest_api_id,
-                                   :resource_id,
-                                   :http_method,
-                                   :authorization
-                                 ])
+    required_variables =
```

Comment:
nit: I feel that this would be simpler if you split it into two different validations. what do you think? Something like this:

```
validate -> { validate_required_attributes([:rest_api_id, :resource_id, :http_method, :authorization]) }

validate -> { validate_required_attributes([:authorization_id]) if  if ['CUSTOM', 'COGNITO_USER_POOL'].include?(:authorization)}
```

Reply:
Also, we shouldn't just support a subset of auth types. We should also support `AWS_IAM` and (maybe) `NONE`

Reply:
https://www.terraform.io/docs/providers/aws/r/api_gateway_method.html#authorization

Reply:
also i think it is `COGNITO_USER_POOLS` (plular), not `COGNITO_USER_POOL` via looking at the docs